### PR TITLE
Add shift-to-invert to remaining directional RGB_* keycode pairs

### PR DESCRIPTION
**Pull request by manna-harbour**
RGB_MODE_FORWARD / RGB_MODE_REVERSE invert their functions when shift is held.
This change adds the same capabilities to the remaining directional RGB_*
keycode pairs.  This improves consistency and provides full RGB control in a
keymap containing only one keycode from each pair.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/quantum.c [hunk 1]
```diff
@@ -354,7 +354,12 @@ bool process_record_quantum(keyrecord_t *record) {
 #    else
             if (!record->event.pressed) {
 #    endif
-                rgblight_increase_hue();
+                uint8_t shifted = get_mods() & (MOD_BIT(KC_LSHIFT) | MOD_BIT(KC_RSHIFT));
+                if (shifted) {
```
Comment (fauxpark) [inline quantum/quantum.c]: There's no need for a variable here, just do `if (get_mods() & MOD_MASK_SHIFT) { ... }` directly :)

Comment (manna-harbour) [reply quantum/quantum.c]: Thanks.  Fixed now, including preexisting occurrences.

File: docs/keycodes.md [hunk 2]
```diff
@@ -389,12 +389,12 @@ This is a reference only. Each group of keys links to the page documenting their
...
-|`RGB_SAI`          |          |Increase saturation                                                 |
-|`RGB_SAD`          |          |Decrease saturation                                                 |
-|`RGB_VAI`          |          |Increase value (brightness)                                         |
-|`RGB_VAD`          |          |Decrease value (brightness)                                         |
+|`RGB_HUI`          |          |Increase hue, decrease hue when Shift is held                       |
+|`RGB_HUD`          |          |Decrease hue, increase hue when Shift is held                       |
+|`RGB_SAI`          |          |Increase saturation, decrease saturation when Shift is held         |
+|`RGB_SAD`          |          |Decrease saturation, increase saturation when Shift is held         |
+|`RGB_VAI`          |          |Increase value (brightness), decrease value when Shift is held      |
+|`RGB_VAD`          |          |Decrease value (brightness), incease value when Shift is held       |
```
Comment (fauxpark) [inline docs/keycodes.md]:
```suggestion
|`RGB_VAD`          |          |Decrease value (brightness), increase value when Shift is held      |
```

File: docs/feature_rgblight.md [hunk 3]
```diff
@@ -48,12 +48,12 @@ Changing the **Value** sets the overall brightness.<br>
...
-|`RGB_SAI`          |          |Increase saturation                                                 |
-|`RGB_SAD`          |          |Decrease saturation                                                 |
-|`RGB_VAI`          |          |Increase value (brightness)                                         |
-|`RGB_VAD`          |          |Decrease value (brightness)                                         |
+|`RGB_HUI`          |          |Increase hue, decrease hue when Shift is held                       |
+|`RGB_HUD`          |          |Decrease hue, increase hue when Shift is held                       |
+|`RGB_SAI`          |          |Increase saturation, decrease saturation when Shift is held         |
+|`RGB_SAD`          |          |Decrease saturation, increase saturation when Shift is held         |
+|`RGB_VAI`          |          |Increase value (brightness), decrease value when Shift is held      |
+|`RGB_VAD`          |          |Decrease value (brightness), incease value when Shift is held       |
```
Comment (fauxpark) [inline docs/feature_rgblight.md]:
```suggestion
|`RGB_VAD`          |          |Decrease value (brightness), increase value when Shift is held      |
```

File: docs/feature_rgb_matrix.md [hunk 4]
```diff
@@ -173,16 +173,20 @@ As mentioned earlier, the center of the keyboard by default is expected to be `{
...
+|`RGB_TOG`          |          |Toggle RGB lighting on or off                                                         |
+|`RGB_MODE_FORWARD` |`RGB_MOD` |Cycle through modes, reverse direction when Shift is held                             |
+|`RGB_MODE_REVERSE` |`RGB_RMOD`|Cycle through modes in reverse, forward direction when Shift is held                  |
+|`RGB_HUI`          |          |Increase hue, decrease hue when Shift is held                                         |
+|`RGB_HUD`          |          |Decrease hue, increase hue when Shift is held                                         |
+|`RGB_SAI`          |          |Increase saturation, decrease saturation when Shift is held                           |
+|`RGB_SAD`          |          |Decrease saturation, increase saturation when Shift is held                           |
+|`RGB_VAI`          |          |Increase value (brightness), decrease value when Shift is held                        |
+|`RGB_VAD`          |          |Decrease value (brightness), incease value when Shift is held                         |
```
Comment (fauxpark) [inline docs/feature_rgb_matrix.md]:
```suggestion
|`RGB_VAD`          |          |Decrease value (brightness), increase value when Shift is held                        |
```

File: docs/keycodes.md [hunk 5]
```diff
@@ -407,19 +407,19 @@ This is a reference only. Each group of keys links to the page documenting their
...
+|`RGB_TOG`          |          |Toggle RGB lighting on or off                                                         |
+|`RGB_MODE_FORWARD` |`RGB_MOD` |Cycle through modes, reverse direction when Shift is held                             |
+|`RGB_MODE_REVERSE` |`RGB_RMOD`|Cycle through modes in reverse, forward direction when Shift is held                  |
+|`RGB_HUI`          |          |Increase hue, decrease hue when Shift is held                                         |
+|`RGB_HUD`          |          |Decrease hue, increase hue when Shift is held                                         |
+|`RGB_SAI`          |          |Increase saturation, decrease saturation when Shift is held                           |
+|`RGB_SAD`          |          |Decrease saturation, increase saturation when Shift is held                           |
+|`RGB_VAI`          |          |Increase value (brightness), decrease value when Shift is held                        |
+|`RGB_VAD`          |          |Decrease value (brightness), incease value when Shift is held                         |
```
Comment (fauxpark) [inline docs/keycodes.md]:
```suggestion
|`RGB_VAD`          |          |Decrease value (brightness), increase value when Shift is held                        |
```

File: docs/keycodes.md (hunk 2 above)
Comment (manna-harbour) [reply docs/keycodes.md]: Thanks!

**Comment by manna-harbour** [comment]
Thanks for catching the typos. Fixed now.

**Comment by fauxpark** [comment]
@manna-harbour there are some conflicts now, due to #7485.

**Comment by manna-harbour** [comment]
> @manna-harbour there are some conflicts now, due to #7485.

Resolved now, thanks!

**Comment by manna-harbour** [comment]
Thanks everyone!

**Pull request merged**